source/clear-linux/guides/maintenance/swupd-mixer-integration.rst

```diff
@@ -0,0 +1,155 @@
+.. _swupd-mixer-integration:
+
+Mixer tool enhancement
+######################
+
+Mixing means composing an operating system for a specific use case. The
+default build of |CLOSIA| provides options to install bundles for various
+server capabilities. If you want to supplement the default OS with
+functionality from other OS distributions, you can manually compose an OS
+using :ref:`the mixer tool <mixer>`.
+
+Software updater release 19900 integrates the mixer tool with the
+updater (:command:`swupd`) and provides a simpler way to mix your OS.
+
+Process overview
+****************
+
+To add a package that does not exist in the current |CL| bundle, perform the following:
+
+#. Create custom packages and bundles using the :file:`swupd-add-pkg` script.
+
+#. Add your desired content with the commands:
+
+   .. code-block:: bash
+
+      sudo swupd update
+      sudo swupd bundle add <new_bundle>
+
+The upstream content is pulled from the normal URL and local content is added
+from a local update store.
+
+.. note::
+
+   The integrated mixer tool adds non-unique packages and replaces upstream
+   content with a local version. However, the updater exits if it finds
+   conflicts before creating the local content. You can override this behavior
+   by passing the :command:`--allow-mix-collisions` option.
+
+
+Prerequisites
+*************
+
+To use the integrated swupd mixer, you need a recent image of
+|CLOSIA| with the mixer bundle installed. If you do not have it already,
+add it with the :command:`swupd bundle-add` command as shown below:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add mixer
+
+
+Using the enhanced swupd mixer
+******************************
+
+Perform the following steps:
+
+#. Create a folder :file:`/usr/share/mix/rpms` for the local update content
+   and copy the desired RPM into the folder:
+
+   .. note::
+
+     The RPM must be formatted and built for |CL|.
+
+   .. code-block:: bash
+
+      sudo cp RPM /usr/share/mix/rpms/
+
+#. Add RPM and create content.
+
+   .. code-block:: bash
+
+      sudo swupd-add-pkg RPM <bundle_name>
+
+   The :file:`swupd-add-pkg` script creates the :command:`<bundle_name>`
+   bundle if it does not exist, adds the RPM to it, and generates the
+   required mixer content and manifests.
+
+#. Use the new :command:`--migrate` flag to let :command:`swupd` know that
+   local content exists.
+
+   .. code-block:: bash
+
+      sudo swupd update --migrate
+
+   When complete, your system is in the mix-ready state using the new
+   Manifest.MoM.
+
```

Comment:
@tmarcu - Is "sudo swupd bundle-list -a" supposed to work after you migrate?  I can see wanting to list available custom bundles.  When I tried it, I got "Failed to retrieve 21130000 MoM manifest" (for example).

Reply:
No it currently does not work because it's not aware of both local and upstream definitions. It's attempting to look upstream only for MoM version 21130000, instead of locally.

Reply:
@bktan8 can we merge this conditionally on fixing this bug if there are no others ?

Reply:
@bktan8 Fix in https://github.com/clearlinux/swupd-client/pull/407

Reply:
@tmarcu - I tested 21150 with known Clear packages (helloworld, joe, nano, emacs) and I haven't been able to add any of these bundles.  I get this error message for each bundle I tried to add:

<bundle> bundle name is invalid, skipping it....
Error: Bad bundle name(s) detected - Aborting

Reply:
Fix upcoming to propagate unchanged custom bundles to the new MoM.

Reply:
https://github.com/clearlinux/swupd-client/pull/409/commits/40d0b60e4e66909b489ea8b344b20f4529bed3a5 Is fixed.

Reply:
@tmarcu - just tested 21360 and still unsuccessful at adding multiple packages/bundles.  
I was able to add the first package.  Subsequent swupd-add-pkg resulted in errors.  See this example:

bktan1@clr-vnc ~ $ sudo swupd-add-pkg helloworld helloworld-bundle
Password: 
mixer is /usr/bin/mixer
/usr/bin/swupd-add-pkg: line 50: [: helloworld: integer expression expected
Initializing mix version 21360010 from upstream version 21360
Adding helloworld to helloworld-bundle...
Directory walk started
Directory walk done - 8 packages
Temporary output repo path: ./.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished
Adding bundle "helloworld-bundle" from local bundles
Bundle "os-core" already in mix; skipping
Building bundles...
=> BUILD BUNDLES
Creating bundles for version 21360010 based on Clear Linux 21360
Preparing new /usr/share/mix/update/image/21360010
  and dnf config: /usr/share/mix/.yum-mix.conf
Packager command-line: dnf --config=/usr/share/mix/.yum-mix.conf -y --releasever=21360
Resolving packages using 4 workers
processing emacs-bundle
processing helloworld-bundle
processing os-core
... done with helloworld-bundle
... done with emacs-bundle
... done with os-core
Resolving files using 4 workers
processing emacs-bundle
processing helloworld-bundle
processing os-core
Bundle helloworld-bundle	263 files
Bundle emacs-bundle	5146 files
Bundle os-core	1957 files
Cleaning DNF cache before full install
Installing all bundles to full chroot
[1/3] emacs-bundle
[2/3] helloworld-bundle
[3/3] os-core
... building special os-core content
... Adding swupd default values to os-core bundle

TIMINGS
  BUILD BUNDLES 32.719s
TOTAL: 32.719s
=> CREATE MANIFESTS
[1/4] emacs-bundle
[2/4] helloworld-bundle
[3/4] os-core
[4/4] full

TIMINGS
  CREATE MANIFESTS 0s
TOTAL: 0s
ERROR: couldn't build update: failed to create update metadata: open /usr/share/mix/update/www/21080/Manifest.big-data-basic: no such file or directory
/home/bktan1
* Downloading 21360 Manifest.MoM...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 11050  100 11050    0     0   138k      0 --:--:-- --:--:-- --:--:--  138k
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5343  100  5343    0     0  70302      0 --:--:-- --:--:-- --:--:-- 71240
* Verified upstream Manifest.MoM
grep: /usr/share/mix/update/www/21360000/Manifest.big-data-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.clr-devops: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.containers-virt-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.desktop-autostart: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.desktop-locales: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.dhcp-server: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.editors: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.emacs: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.haskell-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.java-runtime: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-aws: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-container: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-gce: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-hyperv: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-hyperv-lts: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-kvm: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-clear-containers: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.perl-extras: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.pnp-tools-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.python-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.time-server-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.user-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.vnc-server: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.containers-virt: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.cloud-native-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-hyperv-mini: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-lts: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-pk: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.amateur-radio: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.java-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.java9-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.koji: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-cloudguest: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-cloudguest-aws: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.storage-cluster: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.hardware-bluetooth: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-cloudguest-azure: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.productivity: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.R-extras: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.mail-utils: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-clr-on-clr-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.cloud-control: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.dev-utils-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.hpc-utils: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.letsencrypt-client: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.machine-learning-web-ui: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.mixer: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.network-basic-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.package-utils: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.sysadmin-hostmgmt: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.vim: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.cryptography: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.curl: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.openssh-server: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.openssl: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-cloudguest-vmware: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.php-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.ruby-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.service-os: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.web-server-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.R-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.desktop-apps-extras: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kvm-host: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.libX11client: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.machine-learning-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.network-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.python-extras: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.qt-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.sysadmin-remote: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.tcl-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.telemetrics: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.xfce4-desktop: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.application-server: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.bootloader: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.bootloader-extras: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.c-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.containers-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.containers-basic-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.database-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.database-basic-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.desktop: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.desktop-apps: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.desktop-assets: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.desktop-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.desktop-gnomelibs: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.dev-utils: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.editors-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.games: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.games-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.go-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.go-basic-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.kernel-native: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.mail-utils-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.nodejs-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-cloudguest-gce: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-clr-on-clr: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-installer: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-testsuite: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-testsuite-phoronix: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-utils-gui: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.os-utils-gui-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.performance-tools: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.perl-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.perl-basic-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.pxe-server: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.python-basic-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.python2-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.python3-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.qt-basic-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.rust-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.scm-server: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.shells: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.storage-utils: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.storage-utils-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.stream-server: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.sysadmin-basic: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.sysadmin-basic-dev: No such file or directory
grep: /usr/share/mix/update/www/21360000/Manifest.user-basic-dev: No such file or directory
* Adjusting stats for Manifest.MoM....
M..m	aa868b59ba0816886902b57ce30287535bedd1021c4a7ab396783ce3706ce483	21080	big-data-basic
M..m	fda42b9e9194feb018c865c7980b601d092123b7fbdf06e05c6b2f6b5017515d	21080	clr-devops
M..m	65248998e0e1203eb46b7851812ea5f50e651754fb54ce4b9dbdc92b535ace15	21080	containers-virt-dev
M..m	39be958b03625d0507222996f167de279bc2edaec9a1ff45a86f3cdfac83ca6a	21080	desktop-autostart
M..m	3ac656e9bdb43871f5345cf71c866a67a58d3ce0a2a085efb8e703be4dd3d753	21080	desktop-locales
M..m	1dbd2354eb2cbf47a871a4d70fc5cee0dc0e6df2c940b03ab6d5ac2edbad594d	21080	dhcp-server
M..m	1dfb413eab8bf8e58ae4cddc161c0bc45a9ee697019e6961029712bcdf2bd517	21080	editors
M..m	141e2f712e230e1d72d1de9fc5c71dc0ca5ce9b6eb1076b5af85a87fef2430d2	21080	emacs
M..m	3f71f09fd2aa92a4fa18ed5c74b89c28f290947ed264f0f6e8519ddae2392e65	21080	haskell-basic
M..m	b87272bb31981a786e362bc16948342f02b590b6a47b5cdf5a903f76425a82ec	21080	java-runtime
M..m	158ced7152e8e1a63526f64e290e6af2d0b9d36adfe0857651c0add24bc60da3	21080	kernel-aws
M..m	09f0f1a919a3ef2b71f31cd289ca5a0250b69e842022ec0b1e2c9d397def6885	21080	kernel-container
M..m	c8898acbca3b35a1471a6645bdf0b67ef93d4f7c5a5ec97e2015c2201ed03b16	21080	kernel-gce
M..m	387679c2493cbae98131b663e35f6f5c7b2109968b734086f6c7078ee8d8cb43	21080	kernel-hyperv
M..m	ec207ccf178fd601ee775b6d422e80c56704feed2e925cef0e4accf8a3f7a7cc	21080	kernel-hyperv-lts
M..m	f6478f110a3ca44450a395b8b0f6afd0656219b4b99beed4985316aab742fbfa	21080	kernel-kvm
M..m	0e2fe3d22267dc80f31fce656c9d33e29356635938016727a636426a1c205751	21080	os-clear-containers
M..m	c5dc7dea9eff19a3b8548feeb80f03136773ebf030a77e410950423d519af362	21080	perl-extras
M..m	6264095c6bf2fd7ae8e0e126dfe21c7f53a9e5ea90722e947164f8978e534030	21080	pnp-tools-basic
M..m	923079be29daa7bcc1f20381d4bc64fd698c5441b8ca699669eba3d3f662d26d	21080	python-basic
M..m	8c5d72eabba50610305583556e5c4257608292d09196b1029f7c94d9ddb9d6ae	21080	time-server-basic
M..m	6ebfb74340ce3423519a96611cc1061ef12272f27e67bd46dfbc9d62b8ff44fe	21080	user-basic
M..m	9280d4bd80f621f7897809fd1daf09edc0eb4fe3c56d9b17dc5e63f236d80e2e	21080	vnc-server
M..m	27479b7f313a490efb3d8b07010a4072e67ae2f1ee07a2f07c07b6f34fbdfad6	21100	containers-virt
M..m	b9a1c95d9565f05bc1dcad064460e653c117ef1f11716b23cdcedae92b54e2d1	21210	cloud-native-basic
M..m	407f84000dd42f8cae6d1c53c21401b28d204cec87cf6298285e1e24ec5225c3	21210	kernel-hyperv-mini
M..m	dad03f7224d10596d028e1c5ac61925b44a90e0fb69e26753dd06dba2c100198	21210	kernel-lts
M..m	63b1a43e82013f3a5242121b596b4b0ad091112d97b3fe93f4472d4a04db0d6f	21210	kernel-pk
M..m	0c82f85f27836332fb3079b8cc50ec17ca71ce54bc05e13041b77d50a19c02a5	21260	amateur-radio
M..m	de27f03a4911378b04a922de3708189c377ad834d05d5a21c656cd54209239a2	21260	java-basic
M..m	79c7bb53a8dd9755d6098aef0e9deead7cb1b8af3c4af0fe7a536454f8d1c41c	21260	java9-basic
M..m	4b711607129e1d3930ad1a19280c2b2a3067c83220d69a9de8224b4434f71ab4	21260	koji
M..m	cf14aa99a84a050f7bd2596314ff40128ed042b353e21344a6e532307c5f4418	21260	os-cloudguest
M..m	35bb03cf5d38891473401fa47bc72d769a49ebec471c2c57bf2b5b38e6b0a687	21260	os-cloudguest-aws
M..m	c8126b2699f217a7d62980b0e8cf12db94ed703b0773dc554ea4425dcd1ced53	21260	storage-cluster
M..m	ba7d5c637e98fc861bf775658bed90bb8fc4db9b14953910557a5a996e17d3f7	21280	hardware-bluetooth
M..m	f9efd5c33bb418569df0738696cd8d3836aed736644b05a80d205b37f9566edf	21280	os-cloudguest-azure
M..m	1d780024ca4442e48422ecf1faae711081289300b21a56ca94c030a9c023feb3	21280	productivity
M..m	427911bb073dece11e96db8b590f646e9ab3c61b61c77d7058f9f59e5033ebcd	21290	R-extras
M..m	b79c7688bfd6aa52ff907a630b9d7d312d8424e7fd4c599f0a82c9c7a719a3a6	21290	mail-utils
M..m	be5778d4d1dac15d8503740e2cbf13ad0ef40a5113ed637918b2a2eb389bf1cb	21290	os-clr-on-clr-dev
M..m	99e59bd2935b99b4fb49f62d4b4429799079eca8573ac92c036c90d404f88353	21320	cloud-control
M..m	51cf52313f05fe6a8f94b52661467019bbc6968ffeb282cb54bd1d285173ba24	21320	dev-utils-dev
M..m	369487ea61092f7be801577af0f6fc8a0bd1753eda7638f703bcfacd10cc434d	21320	hpc-utils
M..m	8cfc1571de3008cc8a50c5654ef053ee990bbb4e892f6c5e665c832e7280d48f	21320	letsencrypt-client
M..m	d0ca80a7a9264d344def49e0e2a7b0eb3c6a5199c155c8ff70d8cff88d81fcad	21320	machine-learning-web-ui
M..m	b906cb3a544797918fb4203026e51e20a6f8207dbefabbbcf9629ad897295898	21320	mixer
M..m	44ae549963c7d108cc1c34e302c7726e45c82054733ab30974fc6f116a040c7e	21320	network-basic-dev
M..m	d2a2d2520e8ce6a76401bd0c748367340152cd6238101f5f63a8c1ecfd327eae	21320	package-utils
M..m	cb2fab21597fb90fcb85f3c1be0f2fa3365ad8bf675903ef7a9e115ebd593689	21320	sysadmin-hostmgmt
M..m	d9d670fef00e5530956a03527c892f2a9d12e7a497426491140c388da7d4bb2f	21320	vim
M..m	1baac2da651e056073bb2e924c5467bdd5db326566a97c96e264150704b263ca	21330	cryptography
M..m	4d1b91ce809911dfd1a3e4f239264e048cac6f3037129a1885c0e56896616b5d	21330	curl
M..m	98bb9a5ff158f5822e421ee1969fd9dfa8261f980d8a8c5135034cf6e22b1128	21330	openssh-server
M..m	fced7f64e6191ceea32ec92c6e7ffdd11976e6688533166fd35f26cc6cc5cfd6	21330	openssl
M..m	17f0813210ab836ce310a88e7e7c8b5a5fe5ec77dbef57720168b60b5b5c59fc	21330	os-cloudguest-vmware
M..m	d94b60515eee9856a3139fbcb76c572448d09ecba5aa84e12cefc9e8d390089d	21330	php-basic
M..m	eabbd066ea773c6aff12fd8c17004fc161e656848b73affc103afcd538f2522d	21330	ruby-basic
M..m	03bf1a5f114d69a19c5321f9c473d2a73ec244775a58222d094f0830fe719bf1	21330	service-os
M..m	dc70c8addffa991552e2cc8128011f1881da2cdac0ce1f4664b12db72d14d231	21330	web-server-basic
M..m	d5a45d2d2554b1e5408d9094422db823b388ab7b25f341bc4a6c53f1191d8736	21350	R-basic
M..m	3f912687b29b71366aaecc5db31b12eae6bf4a3335354e649383282a563cdd67	21350	desktop-apps-extras
M..m	9aba744d32f513f74cc207e5d848cc809fa70402f4a73e92bc829f4b6662f3e4	21350	kvm-host
M..m	6f308ca00836da18e5bc209f0e2c464dcba047da3400fa37237ac1b5b974b0f0	21350	libX11client
M..m	6ded6a45071e2e511dbdaf62b07623c3ec026524532e0493864b7003cf36d486	21350	machine-learning-basic
M..m	5b3b3cf3acfb3dc35ff738d2ad2f4d7927167ae2105942bd3264d1763349d066	21350	network-basic
M..m	0d48fe2e4d33c9b45614c0947fdda74639ab7b02224afedb89ceec3c09b41819	21350	os-core-update
M..m	a98344374b2c9599c37d26d31aa0e2c079f54b2ffeefc6eb518d3be96fdae67f	21350	python-extras
M..m	9cf9fecc704c4cba2e4442ee87860db388ae4e4e44e1609ce451fb5b44be4353	21350	qt-basic
M..m	7984454ab2ef308dd406377df1eb2765bcb0d20f794c281290c474c97e107095	21350	sysadmin-remote
M..m	2e72b8be0711b9b9cb2b61183769858e37b80fe35d6286a7cebed5274d373826	21350	tcl-basic
M..m	9889d2f43e0007cacf1bb3d2c1f5e8260b9621ddb10e2cad4f40e716c292acc4	21350	telemetrics
M..m	11bc1e662c11d7bd897c285d55b5af7e8de5daca7eecf24ff37d61e5c384a01f	21350	xfce4-desktop
M..m	5049c20c939813969d92e332aef390ac4ee9d534cc90fe3835f6f90606f8fe5d	21360	application-server
M..m	74298ebf48ac592f586a2972819644f5fbb727644abc7e716e7e0e5d69fffaf9	21360	bootloader
M..m	c249f03876435e048bfd492965cef94a93dddf998028a11d1fc0d8701a5d34e8	21360	bootloader-extras
M..m	dde4cdb26e2fadd75207b36a29ae4e6c12215282e3175222560c4ad5609fbcc4	21360	c-basic
M..m	66c51a0f3eebe1efa297009a3be37c46af9d4a3a8b0033d630ea597e4f81d174	21360	containers-basic
M..m	46bc01e0901f9c6ba676e970b1b3de27421050caa0473f560d117c0fc46b7396	21360	containers-basic-dev
M..m	ce4ce13e6c000500f607ea978997661fe8dbc17c02a77cf408655281c3b71e20	21360	database-basic
M..m	23c40b1385d079da57f28a2cbace2158f3753eda7d028648d6398353a87be790	21360	database-basic-dev
M..m	259b88f73727fe46473c80b513b320b3a1d9cd84df91ac57452cc1d87983fbeb	21360	desktop
M..m	084087786d2f7a8dbec9842ac0900fee253d7c34dcbc6625e2046e54e1a995ab	21360	desktop-apps
M..m	fa2e2ad0436f8009dd3955b79f1d31191e5a816589f36d6509d0a59c78e72b46	21360	desktop-assets
M..m	a7049fce50772c98831f67f9574ed271cc6fea2d87a9785bb5efe643a23bbb86	21360	desktop-dev
M..m	0b2e5fb7d1bb1eb1acca2d06644e6ff0a7280163f01160fb5ad3ac5683cfc5ef	21360	desktop-gnomelibs
M..m	dff7af7844e3ca4b21879128d6222f58380a611fe7a4c6919399dffec1e7f67d	21360	dev-utils
M..m	1b53b27fec41a6923bad0fb60774113c2626450ec207cdbbfa0950b23fed0e4a	21360	editors-dev
M..m	98b3a2aba0714bc354816005d669641f44080a5389a010ea3872331de25fbecf	21360	games
M..m	f84a429b5d69fa7368f24c6c2886303718a631805f67b62f8624f0668dd86ea8	21360	games-dev
M..m	fdb3d126a931d41799204ad5104444cf10c396180922dd6039fe94b6e6bc91aa	21360	go-basic
M..m	a6b832c6ce812e40996d3f7f64a24623952abc1ff5717df6b1ea2e2dfeeb78c1	21360	go-basic-dev
M..m	01180c416fc2e7fba2a2d0ed55f787d02b1a2dd2634979ccce088d8c58af36be	21360	kernel-native
M..m	7d6a163b9c14b2b186a0ee3bc49056ce8f8751bedff3355bb1d035d7b54e9bb7	21360	mail-utils-dev
M..m	f7e9a0ba21ebc5c588551d28ac9d4633642bb11c56219080ed0aa7fbd84f41c3	21360	nodejs-basic
M..m	1b53f7da53d32b083dd4e9bad6cd073e98a3e3e3ecbde2a4592ade0d3f46d13d	21360	os-cloudguest-gce
M..m	6df75cc9f508955680749c4646909c165527a16c1610d26d3ef10928994bd726	21360	os-clr-on-clr
M..m	2e2cf4a8012b72a7c8704a949c9db356e31796ae2108c9a8b6f94e0415ed8c80	21360	os-core-dev
M..m	ac3ffc96a5063b664e6088a62d5e6dcb71da57181b1b7b1f6724cd13e2e7776a	21360	os-core-update-dev
M..m	1bbdf53057bb2d3b26bc3f3afdd836083400da1a666abfea79a101fb71abc0fe	21360	os-installer
M..m	9cf4bcb4379759090f3020031e9576738802d5ac3ee20a5f3f576a5fe0e7600b	21360	os-testsuite
M..m	54a67a7a8c4365bdf3bfe87a2984e0fd25b1182c95e6e5a8825c89d9947e2c28	21360	os-testsuite-phoronix
M..m	fc0d0e57c566163f3d47059195a11572416105baab7d161a3e0ed194cf3a628e	21360	os-utils-gui
M..m	16e0544a307d9e37e2bc3b3ce9860e6db3486bb1f74779c846233dce71916347	21360	os-utils-gui-dev
M..m	d62f243e0b665085f0817059286747f3f87f7fb55371896cade5367a16cba6b6	21360	performance-tools
M..m	e4bbae241131b5ec584d20e48c221b304c259d1695200cfea37202f1ec07d309	21360	perl-basic
M..m	46579138ebf8cf3e2b47be637a2c85c49f3d389a45329bfa4661ac738f58f494	21360	perl-basic-dev
M..m	f039614a6b6d58c9591a2b56b4c00ced4356b556b8eb83d5a33826f6226e040d	21360	pxe-server
M..m	d4f28daf9a40b744f18056cafa6478518e1709758be58b8949190849bbeb13af	21360	python-basic-dev
M..m	14176506f1f4aa03f723a3710d139a3028d0dd21a9376c9b43d7e3d446585749	21360	python2-basic
M..m	8912be113b2eb18ab14d63d3b49204926498a7b9053fcf75e39090d2bb8e3bf2	21360	python3-basic
M..m	3cb5265b41bfd6f3fa8d06765fcbfad5d6e05d3c1f66f4234039a558e0e38ee6	21360	qt-basic-dev
M..m	631db0b8c9755ca608008b5ce85e951b4d3105ad7977e01cd8256ed70fe43d34	21360	rust-basic
M..m	f68cdaa4cfa627a7756aae0d028c2bfa270c69f951bbfadf0ba50c8d11928e47	21360	scm-server
M..m	b46a4b513664f7ba1e60498f6218d87443cc736c6e4d6fdc8f97e7f7f366e08c	21360	shells
M..m	011c5c125415d7415873135ac9edb7c740043b773a5d47b9ac581b788135ebef	21360	storage-utils
M..m	21458ab5fc9de0b42f9c0042a64a8478e24b5df9231141f6a7ec1834ae117035	21360	storage-utils-dev
M..m	c9fd172f0f106432f0ce9b822714a0098450821b8e70493f3d30880c327a1fe8	21360	stream-server
M..m	9177604ff5aa26b400623c4bf157ee53a34c1f196bd9888ba9430d43bac4e7aa	21360	sysadmin-basic
M..m	1c196eb165d60b660aa8b6c57257ccbff5723d6ddf19268ec3f88db4dc0dc937	21360	sysadmin-basic-dev
M..m	ff5d58065a5de1eb57aa40b3cc8f9616401deb4a6de0eabd172ab3efb5957918	21360	user-basic-dev
M..m	a0b98e1441f005f747fbf525ecebad9f28873bf381aa050f7ecf6570345ef655	21360000	emacs-bundle
M..m	0646d0fd6e2273c565bc7bd73130e84d610c1b5feae05fe8c832106d42f74803	21360000	os-core
M..m	494138d89103e5d5419141c74c01f0fb2b4ebcd1d5c4e08cf10a6429531bf34d	21360000	os-core-update-index
Manifest.MoM

Reply:
Then attempt to "migrate" didn't work.  See these errors:

bktan1@clr-vnc ~ $ sudo swupd update --migrate
Password: 
swupd-client software update 3.15.1
   Copyright (C) 2012-2018 Intel Corporation

Attempting to migrate to your mix content...

Update started.
Attempting to download version string to memory
Preparing to update from 21360 to 21360000
Warning: Removing corrupt Manifest.bootloader artifacts and re-downloading...
Failed to load 21360 bootloader manifest
Retry #1 downloading server sub-manifests
Warning: Removing corrupt Manifest.bootloader artifacts and re-downloading...
Failed to load 21360 bootloader manifest
Retry #2 downloading server sub-manifests
Warning: Removing corrupt Manifest.bootloader artifacts and re-downloading...
Failed to load 21360 bootloader manifest
Retry #3 downloading server sub-manifests
Warning: Removing corrupt Manifest.bootloader artifacts and re-downloading...
Failed to load 21360 bootloader manifest
Error: Cannot load server MoM sub-manifests, exiting
Update took 82.3 seconds
bktan1@clr-vnc ~ $ sudo swupd update -s
swupd-client software update 3.15.1
   Copyright (C) 2012-2018 Intel Corporation

Attempting to download version string to memory
Current OS version: 21360
Latest server version: 21360

Reply:
Hold your horses @bktan8 you still have the old version of swupd-client (3.15.1). 3.15.2 hasn't been released yet as of 21360.

Reply:
3.15.2 is out in 21380

Reply:
These exact errors are fixed with the new release of client and mixer.